### PR TITLE
mptcp: add support for MP_TCPRST

### PR DESCRIPTION
**Pull request by dcaratti**
dissect the MP_TCPRST sub-option in accordance to RFC8684 §3.6.

Link: https://github.com/multipath-tcp/mptcp_net-next/issues/189
Signed-off-by: Davide Caratti <dcaratti@redhat.com>

### CODE REVIEW

File: print-mptcp.c
Comment (matttbe) [inline print-mptcp.c:495]: detail: should we indicate `unknown` instead of `invalid`? The RFC specifies 7 possible reasons but it doesn't forbid extensions later.

https://www.rfc-editor.org/rfc/rfc8684.html#name-subflow-reset

File: print-mptcp.c
Comment (matttbe) [inline print-mptcp.c:194]: (very little) detail if you have to update this commit: just to uniform with what is above, maybe best to add `s` (`reasons`) and use `0x0X` (`0x06` instead of `0x6` like in the RFC :)

**Review by matttbe — COMMENTED** [review]
Thanks Davide for working on that!
It looks good to me, just 2 very small details, it would be fine without them but up to you ;-)

File: print-mptcp.c
Comment (dcaratti) [reply print-mptcp.c:495]: indeed, it's not `invalid ` just because it's not a known value of `mp_tcprst_flags`. Will fix that.

File: print-mptcp.c
Comment (dcaratti) [reply print-mptcp.c:194]: sure, will do. Thanks for reviewing!

**Review by matttbe — APPROVED** [review]
thank you for this patch :-)

Looks good to me for the MPTCP part, hopefully to TCPDump maintainers as well :)

File: print-mptcp.c
Comment (fxlb) [inline print-mptcp.c:201]: Thanks for the PR.
There is a missing '{ 0, NULL }' here.
And please rebase on top of the-tcpdump-group:master.

**Review by fxlb — CHANGES_REQUESTED** [review]

File: print-mptcp.c
Comment (dcaratti) [reply print-mptcp.c]: > Thanks for the PR.
> There is a missing '{ 0, NULL }' here.

added, and ran again `make check` - to ensure it still passes when it parses MPTCP_TRST sub-option having `reason` equal to zero. 

> And please rebase on top of the-tcpdump-group:master.

done.
thanks for reviewing!
-- 
davide

**Review by mcr — APPROVED** [review]

**Comment by mcr** [comment]
Will merge when Appveyor checks complete.  They are always slow.

**Review by fxlb — APPROVED** [review]

**Pull request merged**
